.github/workflows/lint-unit.yml

```diff
@@ -92,3 +102,10 @@ jobs:
           echo "::group::Version verification checks"
           ruby ci/check-version-matches.rb
           echo "::endgroup::"
+         
+      - name: Export coverage report
+        uses: actions/upload-artifact@v4
+        if: always()
```

Comment:
As @felixsch mentioned issue regarding storage of artifacts, how about storing the coverage report only in case of failure?

Reply:
This is only for this PR. I will completely remove it before merging it.